stand/efi/loader/framebuffer.c

```diff
@@ -650,17 +651,22 @@ efi_find_framebuffer(teken_gfx_t *gfx_state)
 	gfx_state->tg_fb.fb_bpp = fls(efifb.fb_mask_red | efifb.fb_mask_green |
 	    efifb.fb_mask_blue | efifb.fb_mask_reserved);
 
-	if (gfx_state->tg_shadow_fb != NULL)
-		BS->FreePages((EFI_PHYSICAL_ADDRESS)gfx_state->tg_shadow_fb,
+	/*
+	 * EFI_PHYSICAL_ADDRESS is 8 bytes wide on both 32 and 64-bit
+	 * systems, so we cant just use a pointer.
+	 */
+	tg_shadow_fb = (uintptr_t)gfx_state->tg_shadow_fb;
```

Comment:
Why cast through a uintptr_t when EFI_PHYSICAL_ADDRESS is always 8 bytes? It seems weird.

Reply:
>I don't think I'll push yet as it looks like there's still some discussion to be had.

I'll try to separate the changes that are still being discussed and push after applying the changes you suggested.